src/main/java/studentlist/StudentList.java

```diff
@@ -0,0 +1,315 @@
+package studentlist;
+
+import exceptions.SEPException;
+import exceptions.SEPRangeException;
+import exceptions.SEPNotFoundException;
+import exceptions.SEPFormatException;
+import exceptions.SEPDuplicateException;
+
+import student.Student;
+import ui.UI;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.List;
+
+/**
+ * Manages a list of students and provides methods for adding, deleting, sorting,
+ * printing and validating student data such as student ID, GPA, and preferences.
+ */
+public class StudentList {
```

Comment:
This class is pretty long, you might consider extracting some methods into another helper class and invoke the relevant methods if needed, e.g. ```splitInput```, ```validateStudentId``` etc.